### DISCUSSION TITLE
Notice XML content example

### DISCUSSION THREAD

**Discussion by MathildeDI** [OP-TED/eForms-SDK#768] [Q&A] [answered]
Hello,
I'm triyng to create a new notice using API on preview environnent (https://enotices2.preview.ted.europa.eu/esenders/webjars/swagger-ui/index.html#/Notice/submitNotice)
I'm looking for example of XML content with just the minimal mandatory information. 
I've tried some examples from here but it doesn't seem to work : https://github.com/OP-TED/eForms-SDK/tree/develop/examples/notices
Sorry if the answer is already there somewhere. 
Thank you very much for you help
Mathilde

**Comment by YvesJo** [comment, marked as answer]
Hi,
The set of provided examples also contain non valid XML instances which are clearly identified as such in their name, so first make sure you're using a valid one (validation reports are also here to help you in your selection).
 When using the valid XML instances and trying to submit these, you will definitely face issues if you do not update some information; 

- the submission date (BT-05) to match the actual submission date and time you're submitting the notice,
- the Notice ID and/or version as this notice may have already been submitted

**Reply by MathildeDI** [reply]
Hi, 
Thank you for your help. I'm having an error with the sdk version "No compatible SDK version found for featureVersion=eforms-sdk-1.10" I wonder which sdk version we should us in Preview environement. 
Thanks and Regards,
Mathilde

**Reply by YvesJo** [reply]
Version 1.10 is currently under evaluation, you may use any version from 1.3 to 1.9. Just make sure you use the appropriate example version of the XML instances.

**Reply by MathildeDI** [reply]
I did try 1.9 and it seems to be OK

**Reply by MathildeDI** [reply]
However I still don't undestand. Is the submit API to create a new notice ? If so, how can we know the notice-id when we create a new notice ? I'm looking for a XML example for creating the draft of a new notice.

**Reply by YvesJo** [reply]
The submit API is not to create new notices, it is for notice submission. Any new notice is assigned an identifier (UUID v4, the likelihood to generate two same UUID is almost inexistent), any new version of an existing notice has to have a new version and once valid and gone for publication, no further version is accepted. 
To create a new notice, you may use eNotices2. If you need your own edition solution, then you'll need to develop it and the SDK will help you with that. You may also find an Editor Demo ( [Notice Editor](https://github.com/OP-TED/eforms-notice-editor) which is a proof of concept (i.e. it does not cover everything) and could be a good starting point for your developments.
You may find some documentation on the SDK at https://docs.ted.europa.eu/eforms/latest/index.html

**Reply by MathildeDI** [reply]
Hello,

I did manage to create a new notice with the "submitNotice" API. 

Firstly, I've created a notice manually on https://enotices2.preview.ted.europa.eu/, I've submitted it, and then published it. It works fine and I got the confirmation email. 
Then, I've exported it to get a right xml file, changed the ContractFolderID and notice-id, and some texts too (title...)
Finally, I’ve used that xml file with submitNotice API to create a new notice ; it did work and the notice has been published. 

I wonder, should be not do that ?

Now, if I’m trying to do the same, but, instead of using a published notice xml file as a template, I’m using a validated notice xml file as a template, the notice is created but when I go back to https://enotices2.preview.ted.europa.eu/ I can’t modify it. 
Also, If I’m trying to do the same, using a draft notice xml file as a template, the notice is created but when I go back to https://enotices2.preview.ted.europa.eu/ the status is “Validation failed” which is normal, but it’s not possible to go into the notice to modify and fix it.
Is there a way for a user to modify a notice on eNotices2, after it has been created and initialized with the API ?   

*******************
en français

J’ai finalement réussi à créer un avis de marché avec l’API submitNotice. 
Je suis d’abord allée sur l’interface utilisateur https://enotices2.preview.ted.europa.eu/ pour créer, soumettre et publier un avis de marché. Je l’ai exporté au format XML. 
J’ai utilisé ce modèle XML pour créer un nouveau fichier, en modifiant seulement les identifiants (ContractFolderID and notice-id) et quelques textes. 
J’ai utilisé l’API avec ce fichier et cela a fonctionné ; un nouvel avis de marché a été créé et publié. 

Est-ce que nous ne devons pas utiliser cet API de cette façon ?

Je fais de nouveaux essais : 
1) en prenant pour modèle une notice validée mais non soumise, avec l’API le nouvel avis est créé au statut « validé », mais on ne peut pas, si on retourne sur l’interface utilisateur, le modifier à la main. 
2) en prenant pour modèle une notice créée mais non validée (un brouillon), l’avis est créé au statut « validation échouée », mais on ne peut pas, si on retourne sur l’interface utilisateur, le modifier à la main.
Est-ce qu’il y un moyen de reprendre les modifications à la main (c’est-à-dire par un utilisateur sur l’interface enotices2) d’une notice au préalablement initialisée depuis l’API ?

**Reply by YvesJo** [reply]
Hi Mathilde,
For eNotices 2 questions, please get in touch directly with the helpdesk ted@publications.europa.eu
KR

**Comment by AnnaKSZ** [comment]
Hi Matilde,

I found this for your problem:
"eSenders should only use the API for the submission of notices and refrain from using the User Interface of eNotices2 for this purpose. The output of eNotices2 is not intended to reflect the correct format of notices submitted via API. Likewise, eSenders should not continue a procedure or create a Change notice via the User Interface for a parent notice that was originally sent via the API, and should not use the UI to manage or to import/export notices submitted via API."
source: https://docs.ted.europa.eu/home/eforms/preview/#_enotices2

**Reply by MathildeDI** [reply]
Hello Anna, 
Thank you for the link, it's very clear. 
Regards,
Mathilde
